src/components/contextual/stake/StakePreview.vue

```diff
@@ -49,14 +49,16 @@ const {
   stakeBPT,
   unstakeBPT,
 } = useStaking();
+
+const approvalAmounts = ref([
+  (props.action === 'restake'
+    ? stakedSharesForProvidedPool.value
+    : balanceFor(props.pool.address)
+  ).toString(),
+]);
 const { getTokenApprovalActionsForSpender } = useTokenApprovalActions(
   [props.pool.address],
-  ref([
-    (props.action === 'restake'
-      ? stakedSharesForProvidedPool.value
-      : balanceFor(props.pool.address)
-    ).toString(),
-  ])
+  approvalAmounts
```

Comment:
Ok, one more small refactor, I think this ref should be a computed property and only provide the single approval value relevant for the pool address. Something like this:
```ts
const approvalAmount = computed((): string => {
  return (props.action === 'restake'
    ? stakedSharesForProvidedPool.value
    : balanceFor(props.pool.address)
  ).toString()
});

const { getTokenApprovalActionsForSpender } = useTokenApprovalActions(
  [props.pool.address],
  [approvalAmount.value]
);

Reply:
Ok, we can reuse `numSharesToModify` for that.